File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    len_clothes = len(clothes)
    types = []
    for i in range(len_clothes):
        if clothes[i][1] not in types:
            types.append(clothes[i][1])
    len_types = len(types)
    new_clothes = [[] for _ in range(len_types)]
    for i in range(len_clothes):
        new_clothes[types.index(clothes[i][1])].append(clothes[i][0])
    
    answer = 1
    for i in range(len_types):
        answer *= (len(new_clothes[i])+1)
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let new_clothes = {}
    
    for (let i = 0; i < clothes.length; i++) {
        let type = clothes[i][1]
        if (!new_clothes[type]) {
            new_clothes[type] = []
        }
        new_clothes[type].push(clothes[i][0])
        }
            
    let answer = 1
    for (let type in new_clothes) {
        answer *= (new_clothes[type].length + 1)
    }
    return answer - 1
}